### ISSUE TITLE
update pinv calls

### ISSUE THREAD

**Issue by lukeolson** [pyamg/pyamg#218]
The `pinv` calls use low level lapack wrappers.  There are several issues with this:

1. difficulty to maintain #216, #209, #213 
2. speed -- we make a claim that it's faster (see below)
3. code bloat

This https://gist.github.com/lukeolson/529c010c475fdaca9e741ddc2dba3b43

![test](https://user-images.githubusercontent.com/1615863/35699170-e4031694-0754-11e8-91a8-f0677d541ab3.png)

Suggests simply using `numpy.linalg.pinv`

@jbschroder what do you think

**Comment by jbschroder** [comment]
Hey Luke,

What's the X-axis in your plot?  Is that the system size, or the number of
systems being pseudo-inverted?

I found (years ago) that the lapack wrappers were faster for inverting a
lot of very small systems, say inverting 10K systems of size 4x4.  The
issue is that looping over a bunch of really small matrices in Python is
slow, but that using the lapack wrappers was able to speed things up.

I just ran a little test, and that's at least still the case for my
laptop.  PyAMG's pinv_array is about 5x faster.  That said, it is a pain to
maintain.

In [4]: from pyamg import *
In [5]: from numpy import linalg
In [6]: from numpy import *
In [9]: from pyamg import util
In [12]: def my_pinv_array(a):
    ...:     for i in range(a.shape[0]):
    ...:         a[i,:,:] = linalg.pinv(a[i,:,:])
    ...:
In [14]: from scipy import rand
In [15]: a = rand(10000,4,4)

In [17]: %timeit -n 10  my_pinv_array(a)
10 loops, best of 3: 366 ms per loop

In [18]: %timeit -n 10  util.linalg.pinv_array(a)
10 loops, best of 3: 75.6 ms per loop

In [19]: a = rand(100000,4,4)


In [20]: %timeit -n 10  my_pinv_array(a)
10 loops, best of 3: 3.79 s per loop

In [21]: %timeit -n 10  util.linalg.pinv_array(a)
10 loops, best of 3: 783 ms per loop
